java/core/lance-jni/src/error.rs

```diff
@@ -15,149 +15,171 @@
 use std::str::Utf8Error;
 
 use arrow_schema::ArrowError;
-use jni::errors::Error as JniError;
+use jni::{errors::Error as JniError, JNIEnv};
+use lance::error::Error as LanceError;
 use serde_json::Error as JsonError;
-use snafu::{Location, Snafu};
 
-/// Java Exception types
-pub enum JavaException {
+pub type JavaResult<T> = std::result::Result<T, JavaError>;
+
+#[derive(Debug)]
+pub enum JavaExceptionClass {
     IllegalArgumentException,
     IOException,
     RuntimeException,
+    UnsupportedOperationException,
 }
 
-impl JavaException {
+impl JavaExceptionClass {
     pub fn as_str(&self) -> &str {
         match self {
             Self::IllegalArgumentException => "java/lang/IllegalArgumentException",
             Self::IOException => "java/io/IOException",
             Self::RuntimeException => "java/lang/RuntimeException",
+            Self::UnsupportedOperationException => "java/lang/UnsupportedOperationException",
         }
     }
 }
 
-#[derive(Debug, Snafu)]
-#[snafu(visibility(pub))]
-pub enum Error {
-    #[snafu(display("JNI error: {message}, {location}"))]
-    Jni { message: String, location: Location },
-    #[snafu(display("Invalid argument: {message}, {location}"))]
-    InvalidArgument { message: String, location: Location },
-    #[snafu(display("IO error: {message}, {location}"))]
-    IO { message: String, location: Location },
-    #[snafu(display("Arrow error: {message}, {location}"))]
-    Arrow { message: String, location: Location },
-    #[snafu(display("Index error: {message}, {location}"))]
-    Index { message: String, location: Location },
-    #[snafu(display("JSON error: {message}, {location}"))]
-    JSON { message: String, location: Location },
-    #[snafu(display("Dataset at path {path} was not found, {location}"))]
-    DatasetNotFound { path: String, location: Location },
-    #[snafu(display("Dataset already exists: {uri}, {location}"))]
-    DatasetAlreadyExists { uri: String, location: Location },
-    #[snafu(display("Unknown error: {message}, {location}"))]
-    Other { message: String, location: Location },
+#[derive(Debug)]
+pub struct JavaError {
```

Comment:
Lets just name this to Error?

Reply:
This is to mimic PyResult and PyErr lol
represent a special kind of end result/error